lib/arel/visitors/to_sql.rb

```diff
@@ -166,6 +166,28 @@ def visit_Arel_Nodes_False o, collector
         collector << "FALSE"
       end
 
+      def visit_Arel_Nodes_ValuesList o, collector
+        collector << "VALUES "
+
+        len = o.rows.length - 1
+        o.rows.each_with_index { |row, i|
+          collector << '('
+          row_len = row.length - 1
+          row.each_with_index do |value, k|
```

Comment:
I have some idea how to refactor it but I don't think it's in the scope of this PR.